code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -372,11 +373,39 @@
 		CB.update_icon()
 		playsound(get_turf(CB), 'sound/weapons/gun_interactions/shotgun_fall.ogg', 70, 1)
 		num_unloaded++
+
+	if(sleight_of_handling(user))
+		return
+
 	if(num_unloaded)
 		to_chat(user, "<span class = 'notice'>You break open \the [src] and unload [num_unloaded] shell\s.</span>")
 	else
 		to_chat(user, "<span class='notice'>[src] is empty.</span>")
 
+/obj/item/gun/projectile/revolver/doublebarrel/proc/sleight_of_handling(mob/living/carbon/human/user)
+	if(!istype(get_area(user), /area/station/service/bar))
+		return FALSE
+	if(!istype(user) || !HAS_TRAIT(user, TRAIT_SLEIGHT_OF_HAND))
+		return FALSE
+	if(!istype(user.belt, /obj/item/storage/belt/bandolier))
+		return FALSE
+	var/obj/item/storage/belt/bandolier/our_bandolier = user.belt
+
+	var/loaded_shells = 0
```

Comment:
Shouldn't this be the amount of shells currently stored in the gun? Or is this only usable on empty guns and I missed that

Reply:
loaded_shells is incremented for every shell that's loaded into the gun by the trait, and it's then shown to the player in the chat message
but that's all after any shells that were in the gun are unloaded in /obj/item/gun/projectile/revolver/doublebarrel/attack_self(), so it should only be empty guns after that

Reply:
Gotcha, forgot about double barrelled shotguns doing that